### PR TITLE
More art-suite recipe improvements

### PR DESCRIPTION
**Pull request by greenc-FNAL**
- **Art suite dependency improvements**
- **[canvas] `hep-concurrency` dependency -> build & test**
- **New versions for art-suite 3.14.04/critic-suite 2.13.06**

### CODE REVIEW

File: packages/cetlib/package.py [hunk 1]
```diff
@@ -46,14 +46,17 @@ class Cetlib(CMakePackage):
...
     depends_on("perl", type=("build", "run"))
     depends_on("sqlite@3.8.2:")
-    depends_on("catch2@3:", when="@3.17:", type=("build", "test"))
-    depends_on("catch2@2.3.0:", when="@:3.16.99", type=("build", "test"))
-    depends_on("catch2", type=("build", "test"))
+    depends_on("catch2@3.3.0:", when="@3.17:", type=("build", "test"))
+    depends_on("catch2@2.3.0:2", when="@:3.16", type=("build", "test"))
     depends_on("cetmodules", type="build")
     conflicts("cetmodules@:3.21.00", when="catch2@3:")
+    depends_on("tbb")
```
Comment (knoepfel) [inline packages/cetlib/package.py:59]: Is `tbb` not just a `build` or `test` dependency?

Comment (greenc-FNAL) [reply packages/cetlib/package.py:59]: It shouldn't be any kind of dependency, but it was added for a reason, and now I'm very unsure what that reasoning was. Checking now...

Comment (marcmengel) [reply packages/cetlib/package.py:59]: Lots of these dependencies were added due to CMake being unable to find things / header files not being able to find other header files, and probably should have been build dependencies. (although if you need the header, you may really need the library, so...)

**Review by marcmengel — APPROVED** [review]

**Comment by knoepfel** [comment]
Ack! I was hoping we'd understand the `tbb` issue first before merging this.  `cetlib` is close to the bottom of the world, so I'd like to not introduce more dependencies there than we need to.

**Pull request merged**